### PR TITLE
added enter allocation proposal functionality to ccd configuration

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPL-13

### Change description ###

Added enter allocation proposal functionality to ccd configuration. added tests to click hyperlinks, go to a page and return to the application, as well as fill out information in form. Created helper function for browser back to facilitate this. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
screenshot_2018-10-11 core case data hmcts 1

File: e2e/steps_file.js [hunk 1]
```diff
@@ -26,5 +26,12 @@ module.exports = function () {
     seeEventSubmissionConfirmation(event) {
       this.see(`updated with event: ${event}`);
     },
+
+    clickHyperlinkAndReturnToOriginalPage(link, pageGoingTo, originalPage) {
```
Comment (geek-soft) [inline e2e/steps_file.js]: I would probably remove instructions that take users back from this function. This is to keep functions single responsibility.

File: e2e/paths/enterAllocationProposal_test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const config = require('../config.js');
+
+Feature('EnterAllocationProposal');
```
Comment (geek-soft) [inline e2e/paths/enterAllocationProposal_test.js]: Let's use plain English here.

File: e2e/pages/enterAllocationProposal/enterAllocationProposal.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  selectAllocationProposal(proposal) {
+    I.waitForElement(this.fields.allocationProposal);
+    I.selectOption(this.fields.allocationProposal, proposal);
+  },
+
+  enterProposalReason(reason) {
+    I.fillField(this.fields.proposalReason, reason);
+  },
+
+  clickDocumentLink(document) {
```
Comment (geek-soft) [inline e2e/pages/enterAllocationProposal/enterAllocationProposal.js]: That appears to be not used.

File: e2e/steps_file.js (hunk 1 above)
Comment (swalker125) [reply e2e/steps_file.js]: Fine by me

File: e2e/paths/enterAllocationProposal_test.js (hunk 2 above)
Comment (swalker125) [reply e2e/paths/enterAllocationProposal_test.js]: How codecptjs configures the file when you generate test files from the command line, shall fix

File: e2e/pages/enterAllocationProposal/enterAllocationProposal.js (hunk 3 above)
Comment (swalker125) [reply e2e/pages/enterAllocationProposal/enterAllocationProposal.js]: You're right, was replaced with the function from the steps_file

**Pull request merged**